src/linter/ui5Types/BestPractices.ts

```diff
@@ -0,0 +1,271 @@
+}
+
+export default function analyzeComponentJson(
+	node: ts.ExpressionWithTypeArguments,
+	manifestContent: string | undefined,
+	reporter: SourceFileReporter,
+	checker: ts.TypeChecker
+) {
+	let parent = node.parent;
+	let classDesc;
+	while (!parent || parent.kind !== ts.SyntaxKind.SourceFile) {
+		if (parent.kind === ts.SyntaxKind.ClassDeclaration) {
+			classDesc = parent;
+		}
+		parent = parent.parent;
+	}
+
+	if (!ts.isSourceFile(parent) || !parent.fileName.endsWith("/Component.js") || !classDesc) {
+		return;
+	}
+
+	if (classDesc && ts.isClassDeclaration(classDesc)) {
+		const analysisResult = findAsyncInterface(classDesc, manifestContent, checker);
+
+		if (analysisResult) {
+			reportResults(analysisResult, reporter, classDesc);
+		}
+	}
+}
+
+function mergeResults(a: AsyncInterfaceFindType, b: AsyncInterfaceFindType): AsyncInterfaceFindType {
+	// null = parent property does not exist i.e. rootView
+	// undefined = async flag is missing
+	// true|false = async flag is explicitly set
+
+	const compareValues = (aProp: null | undefined | boolean,
+		bProp: null | undefined | boolean): null | undefined | boolean => {
+		let result = null;
+		if (aProp === undefined || bProp === undefined) {
+			result = undefined;
+		}
+		if (typeof aProp === "boolean" || typeof bProp === "boolean") {
+			// eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+			result = aProp || bProp;
+		}
+
+		return result;
+	};
+
+	return {
+		routingAsyncFlag: compareValues(a.routingAsyncFlag, b.routingAsyncFlag),
+		rootViewAsyncFlag: compareValues(a.rootViewAsyncFlag, b.rootViewAsyncFlag),
+		hasAsyncInterface: compareValues(a.hasAsyncInterface, b.hasAsyncInterface),
+	};
+}
+
+function findAsyncInterface(
+	classDefinition: ts.ClassDeclaration,
+	manifestContent: string | undefined,
+	checker: ts.TypeChecker
+): AsyncInterfaceFindType | undefined {
+	if (ts.isClassDeclaration(classDefinition)) {
+		const returnTypeTemplate = {
+			hasAsyncInterface: null,
+			routingAsyncFlag: null,
+			rootViewAsyncFlag: null,
+		} as AsyncInterfaceFindType;
+
+		// Checks the interfaces and manifest
+		const curClassAnalysis = classDefinition.members.reduce((acc, member) => {
+			const checkResult = doChecks(member as ts.PropertyDeclaration, manifestContent);
+			return mergeResults(acc, checkResult);
+		}, {...returnTypeTemplate});
+
+		const heritageAnalysis =
+			classDefinition?.heritageClauses?.flatMap((parentClasses: ts.HeritageClause) => {
+				return parentClasses.types.flatMap((parentClass) => {
+					const parentClassType = checker.getTypeAtLocation(parentClass);
+
+					return parentClassType.symbol?.declarations?.flatMap((declaration) => {
+						let result = {...returnTypeTemplate} as AsyncInterfaceFindType;
+						// Continue down the heritage chain to search for
+						// the async interface or manifest flags
+						if (ts.isClassDeclaration(declaration) &&
+							declaration?.name?.getText() !== "UIComponent") {
```

Comment:
What about a `mycomp.base.UIComponent` class that itself inherits from another class? Isn't this check a bit too vague, as it only checks for the class name, but not in which namespace this class exists?

Reply:
Done with https://github.com/SAP/ui5-linter/pull/73/commits/423d76671e8632ffdd824e50d2233678257bf9be